docs/adr/adr_authorization-specification.md

```diff
@@ -0,0 +1,173 @@
+
+#### Usergroups
+
+A usergroup reflects the user's responsibilities in the partner organisations. Currently, these usergroups exist:
+
+- administrator (head of operations)
+- coordinator
+- warehouse volunteer
+- free shop volunteer
+- library volunteer
+- label creator
+- God user
+
+#### Permissions
+
+Depending on the user group, a user is able to perform certain actions. In our authorization concept, these actions are mirrored by action-based permissions (ABP).
+ABPs represent functions in the application that the user who was granted the ABPs can execute. Examples: viewing the inventory, or managing products. The name of an ABP consists of a verb indicating the action, and a plural noun (if applicable), separated by underscore: `manage_products`, `view_inventory`, etc.
+
+In the application back-end however it needs to be distinguished which data a user is allowed to access in what way. This is achieved through resource-based permissions (RBP).
+An RBP refers to a resource in the database, and the methods that the user who was granted the RBP can execute. These methods correspond to database operations:
+
+- read: `SELECT`
+- create: `INSERT`
+- edit: `UPDATE`
+- write: `INSERT`, `UPDATE`
+- delete: `DELETE`
+- assign: `INSERT` into cross-reference table
+
+The naming convention for RBP is a singular noun (the resource; multi-word nouns concatenated by underscore), and a method name, separated by colon: `user:edit`, `beneficiary:create`, `tag_relation:read`, etc.
+
+Every ABP comprises one or more RBP, e.g. the ABP `manage_tags` stands for `tag:write`, `stock:read`, `tag_relation:read`, and `beneficiary:read`.
+
+The mapping of usergroup to ABPs, and ABP to RBPs is listed in [this document](https://docs.google.com/spreadsheets/d/1W4YWcc59wUFUWgReumdH6DQ4zU7JcTgvf6WEbdqaGHQ/edit#gid=0).
+
+### User management in Auth0
+
+[Auth0](https://auth0.com) is a service for managing user authentication and authorization. It serves as single source of truth. An authenticated user gets issued a JSON Web Token (JWT) holding their authorization information. For signing the JWT we use the RS256 algorithm: the token will be signed with our private signing key and can be verified using our public signing key.
+
+#### Reasons to use Auth0
+
+Cf. [related ADR](./docs/adr/adr_auth0.md)
+
+#### User entity in Auth0
+
+Any user registered for boxtribute has their authorization data (`app_metadata`) stored in the Auth0 database. The user attributes are:
+
+- `is_god`: `"1"` for god user, `"0"` otherwise
+- `usergroup_id`: the ID of the usergroup the user belongs to
+- `base_ids`: a list of base IDs that the user has access to
+- `organisation_id`: the ID of the organisation the user belongs to
+
+During registration, the user gets assigned a role, indicating their usergroup and the bases they belong to. The role is named like `base_1_coordinator`.
+
+From the user's role the ABPs are derived via a script (?).
```

Comment:
@vahidbazzaz I'm not certain how the ABPs (Auth0 permissions) are concluded, please clarify :)

Reply:
generally yes. There is just one script called [`create-dynamic-permissions`](https://github.com/boxwise/system-management/blob/main/services/auth0/dev/actions/create-dynamic-permissions/code.js).
Based on the roles it adds ABPs to the id token and RBPs to the access_token after the login

Reply:
@pylipp Yep, we have them inside Auth0 script as a source of trust

Reply:
Ah, does this mean that the Auth0 permissions assigned to a user don't have any effect at all? @vahidbazzaz @HaGuesto 
It's the list in the 'Permissions' tab of a user's details:
![2022-11-02-090425_1053x464_scrot](https://user-images.githubusercontent.com/10617122/199433534-b8de580b-cbe1-4a65-9812-1594eb0d513a.png)

Reply:
@pylipp yes atm only role itself take into the account and inside the Auth0 action script the permissions being applied to the JWT payload. This can be changed in the future if we want to have customized permissions per user, but its add some extra complexity to the Auth0 action script and also there are some limitations on that end as well.